packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -271,3 +281,105 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
     return result;
   }
 }
+
+class _CupertinoPickerSemantics extends SingleChildRenderObjectWidget {
+  const _CupertinoPickerSemantics({
+    @required this.controller,
+    Widget child,
+  }) : super(child: child);
+
+  final FixedExtentScrollController controller;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoPickerSemantics(controller, Directionality.of(context));
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderCupertinoPickerSemantics renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..controller = controller;
+  }
+}
+
+class _RenderCupertinoPickerSemantics extends RenderProxyBox {
```

Comment:
Somewhere (maybe here or on assembleSemanticsNode) it should be documented what this class is attempting to do to make life easer for whomever comes across this next.